lms/djangoapps/courseware/views/views.py

```diff
@@ -444,6 +452,60 @@ def get_cosmetic_display_price(course, registration_price):
         return _('Free')
 
 
+class EnrollStaffView(View):
+    """
+        Determine If user is global staff, it will be redirected to page "enroll_satff.html"
+        that asks you if you want to register for the course.
+        This pages has the courseware link you were heading to as a ?next parameter
+        Click "enroll" and be redirected to the page you wanted to go to
+        Click "Don't enroll" and go back to the course about page
+
+    Arguments:
+     - request    : HTTP request
+     - course_id  : course id
+
+    Returns:
+     -RedirectResponse
+    """
+    template_name = 'enroll_staff.html'
+
+    @method_decorator(require_global_staff)
+    @method_decorator(ensure_valid_course_key)
+    def get(self, request, course_id):
+        """
+        Renders Enroll Staff View
+        """
+        user = request.user
+        course_key = CourseKey.from_string(course_id)
+        with modulestore().bulk_operations(course_key):
+            course = get_course_with_access(user, 'load', course_key)
+            if not registered_for_course(course, user):
+                context = {'course': course,
+                           'csrftoken': csrf(request)["csrf_token"]}
+                return render_to_response(self.template_name, context)
+
+    @method_decorator(require_global_staff)
+    @method_decorator(ensure_valid_course_key)
+    def post(self, request, course_id):
+        """
+        Enroll and returns the response
+        """
+        _next = urllib.quote_plus(request.GET.get('next', 'info'), safe='/:?=')
+        course_key = CourseKey.from_string(course_id)
+        if 'enroll' in request.POST:
```

Comment:
question (I don't know the "right" way to do this), but, should the data here be `{"enroll": True}` or `{"enroll": False}` instead of `{"enroll": "Enroll"}` or `{"dont_enroll": "Don't Enroll"}`? That way you'd rewrite this like:

```python
enroll = request.POST.get('enroll', False)
if enroll:
    ...
else:
    ...
```